woocommerce/wc-admin
# Add data source filter to remote inbox notification system

Adds a filter to allow easier testing of the remote inbox system with a custom data source.

Before merging, I think it's worth checking if there are reasons to not implement this (mostly around potential abuse). cc @becdetat 

### Detailed test instructions:

1. Add a plugin that filters the data sources:
```php
add_filter(
    'woocommerce_admin_remote_inbox_data_sources',
    function() {
        return array( 'http://mysite.com/data-source.json' );
    }
);
```
2. Add some data to the source (`data-source.json` in the above example).
```json
[
    {
        "slug": "test-poll-sources",
        "type": "info",
        "status": "unactioned",
        "is_snoozable": 0,
        "source": "local",
        "locales": [
            {
                "locale": "en_US",
                "title": "Test Poll Sources",
                "content": "I'm a remote note from a filtered source!"
            }
        ],
        "actions": [],
        "rules": []
    }
]
```
3. Run the `wc_admin_daily` cron event.
4. Check that your data was successfully added to `wc_remote_inbox_notifications_specs`.
